### PR TITLE
Dark Theme Added

### PR DESCRIPTION
**Pull request by jeetbhalani15**
## Added  Theme Provider For Dark Theme
- Added toggle button for themes
- Added Theme provider

### CODE REVIEW

**Review by Jay-Gadhiya — APPROVED** [review]
-code looks clean
-ui looks good
- toogle button is working fine

**Comment by hussiantabha** [comment]
Everything working fine and clean UI. Good to merge.

File: Gravity-Notes/src/Components/Navigation/Navbar.css [hunk 1]
```diff
@@ -30,6 +30,9 @@
     font-family: monospace;
     font-weight: 600;
 }
+.logo-txt{
+    color:grey;
```
Comment (JuvitaSaini09) [inline Gravity-Notes/src/Components/Navigation/Navbar.css:34]: suggesstoin:use either hex or simple color name just for consistency

File: Gravity-Notes/src/Components/Navigation/Navbar.css [hunk 2]
```diff
@@ -82,15 +85,15 @@
     border-radius: 50%;
     height: 52px;
```
Comment (JuvitaSaini09) [inline Gravity-Notes/src/Components/Navigation/Navbar.css:86]: use rem

File: Gravity-Notes/src/Components/Navigation/Navbar.css [hunk 3]
```diff
@@ -82,15 +85,15 @@
...
     border: none;
-    background-color: #ffffff;
+    background: transparent;
 }
 .widgets2 button{
     padding: 12px;
     font-size: large;
     border-radius: 50%;
     height: 52px;
```
Comment (JuvitaSaini09) [inline Gravity-Notes/src/Components/Navigation/Navbar.css:94]: use rem

**Review by JuvitaSaini09 — APPROVED** [review]
looks good to merge

File: Gravity-Notes/src/Components/Navigation/Navbar.js [hunk 4]
```diff
@@ -12,17 +12,27 @@ import {Link} from "react-router-dom";
...
+import { FaSun } from 'react-icons/fa';
 
 
 
 export const Navbar = ()=>{
   const {authState} = useAuth();
+  const {darkTheme , setDarkTheme} = useTheme();
 
   const logOutHandler =()=>{
     console.log(authState.token)
```
Comment (keyurshah339) [inline Gravity-Notes/src/Components/Navigation/Navbar.js:28]: remove all the console logs

File: Gravity-Notes/src/Components/Navigation/Navbar.css [hunk 5]
```diff
@@ -63,7 +66,7 @@
     font-size: large;
     border-radius: 15px;
     border: none;
-    background-color: #F1F3F4;
+    /* background-color: #F1F3F4; */
```
Comment (ark2002) [inline Gravity-Notes/src/Components/Navigation/Navbar.css:69]: remove comments before commiting

File: Gravity-Notes/src/Components/Navigation/Navbar.css (hunk 3 above)
Comment (ark2002) [inline Gravity-Notes/src/Components/Navigation/Navbar.css:94]: give a proper fontsize

File: Gravity-Notes/src/Components/Navigation/Navbar.css [hunk 6]
```diff
@@ -109,7 +112,8 @@
     font-size: inherit;
     border-radius: 50%;
     border: none;
-    background-color: #ffffff;
+    background: transparent;
+    /* background-color: #ffffff; */
 }
```
Comment (ark2002) [inline Gravity-Notes/src/Components/Navigation/Navbar.css:117]: same comments here  aswell

File: Gravity-Notes/src/Components/Navigation/Navbar.css [hunk 7]
```diff
@@ -156,7 +161,7 @@ a:hover {
     height: 56px;
     width: 56px;
```
Comment (ark2002) [inline Gravity-Notes/src/Components/Navigation/Navbar.css:162]: use rems where u can

File: Gravity-Notes/src/Components/Navigation/Navbar.js [hunk 8]
```diff
@@ -12,17 +12,27 @@ import {Link} from "react-router-dom";
 
 import "./Navbar.css";
 import { useAuth } from '../../Contexts/Auth-context';
+import { BsToggleOn } from 'react-icons/bs';
+import { useTheme } from '../../Contexts/Theme-context';
```
Comment (ark2002) [inline Gravity-Notes/src/Components/Navigation/Navbar.js:16]: use barrel exports in context folder

File: Gravity-Notes/src/Components/Notes-card/Notecard.css [hunk 9]
```diff
@@ -40,6 +40,9 @@
     border-radius: 50%;
     height:30px;
 }
+.label-txt{
+    color: white;
+}
```
Comment (ark2002) [inline Gravity-Notes/src/Components/Notes-card/Notecard.css:45]: use variables for reoccurring colors

File: Gravity-Notes/src/Contexts/StateProvider.js [hunk 10]
```diff
@@ -1,14 +1,17 @@
 import { AuthProvider } from "./Auth-context";
 import { NotesProvider } from "./NotesAction-context";
+import { ThemeProvider } from "./Theme-context";
```
Comment (ark2002) [inline Gravity-Notes/src/Contexts/StateProvider.js:3]: barrel export would had been efficient here

File: Gravity-Notes/src/Pages/Homepage/Homepage.css [hunk 11]
```diff
@@ -74,7 +74,7 @@
     }    
  }
 
- @media screen and (max-width:430px) {
+ @media screen and (max-width:429px) {
```
Comment (ark2002) [inline Gravity-Notes/src/Pages/Homepage/Homepage.css:79]: use rems for such big values

File: Gravity-Notes/src/Pages/Landing-page/Landing-page.js [hunk 12]
```diff
@@ -26,10 +26,10 @@ return(
                         <li className="nav-btn-txt">Home</li>
                         <li className="nav-btn-txt">Docs</li>
                         <li className="nav-btn-txt">About Us</li>
-                        <li><button className="toggle-btn">
+                        {/* <li><button className="toggle-btn">
                             <i className="fa fa-moon-o"></i>
                             <i className="fa fa-sun-o"></i>
-                          </button></li>
```
Comment (ark2002) [inline Gravity-Notes/src/Pages/Landing-page/Landing-page.js:32]: remove comments before commiting

File: Gravity-Notes/src/Pages/Logout/Logout.js [hunk 13]
```diff
@@ -3,9 +3,11 @@ import { Link } from "react-router-dom";
 import { Footer } from "../../Components/Footer/Footer";
 import { Navbar } from "../../Components/Navigation/Navbar";
 import { Sidebar } from "../../Components/Sidebar/Sidebar";
```
Comment (ark2002) [inline Gravity-Notes/src/Pages/Logout/Logout.js:5]: barrel exports would help

**Review by ark2002 — APPROVED** [review]
some changes

File: Gravity-Notes/src/Components/Navigation/Navbar.js (hunk 8 above)
Comment (jeetbhalani15) [reply Gravity-Notes/src/Components/Navigation/Navbar.js:16]: ok i will add

File: Gravity-Notes/src/Components/Notes-card/Notecard.css (hunk 9 above)
Comment (jeetbhalani15) [reply Gravity-Notes/src/Components/Notes-card/Notecard.css:45]: ok

File: Gravity-Notes/src/Pages/Homepage/Homepage.css (hunk 11 above)
Comment (jeetbhalani15) [reply Gravity-Notes/src/Pages/Homepage/Homepage.css:79]: its media screen it should always be in px ig

File: Gravity-Notes/src/Components/Sidebar/Sidebar.js [hunk 14]
```diff
@@ -33,26 +35,26 @@ export const Sidebar = ()=>{
...
+                <span  className={`tag ${darkTheme ? "light-txt" : null}`}>Array</span>
             </li>
     
             <li className="item unused-btn">
                 <button className="sidebar-btn" ><a href="#"><BiTagAlt size={30}/></a></button>
-                <span  className="tag">String</span>
+                <span  className={`tag ${darkTheme ? "light-txt" : null}`}>String</span>
             </li>
             <li className="item">
                 <Link to="/label"><button className="sidebar-btn" ><a href="#"><BsTags size={30}/></a></button>
```
Comment (keyurshah339) [inline Gravity-Notes/src/Components/Sidebar/Sidebar.js:55]: why using anchor tags?

File: Gravity-Notes/src/Pages/AuthWidget/Logout/Logout.css [hunk 15]
```diff
@@ -29,4 +29,7 @@
     color: #fdfdfd;
     text-decoration-line: underline;
     font-size: large;
+}
+.logout-txt{
+    color: white;
```
Comment (keyurshah339) [inline Gravity-Notes/src/Pages/AuthWidget/Logout/Logout.css:34]: user color varibales

**Pull request merged**